src/foam/nanos/ruler/RuleEngine.java

```diff
@@ -196,7 +197,13 @@ private void applyAsyncRule(Rule rule, FObject obj, FObject oldObj) {
       if ( ! rule.f(x, nu, oldObj) ) return;
 
       PM pm = PM.create(getX(), RulerDAO.getOwnClassInfo(), "ASYNC", rule.getDaoKey(), rule.getId());
+      var before = nu.fclone();
       rule.asyncApply(x, nu, oldObj, RuleEngine.this, rule);
+      if ( rule.getOperation() != Operation.REMOVE
```

Comment:
No use doing the fclone() above if this is a REMOVE operation.

Reply:
Why do a diff() rather than just equals()?

Reply:
this is NOT a remove operation. He needs to fclone so he can compare the two objects later on. What is the difference between diff and equals?

Reply:
equals() is much faster because it just returns a boolean of if they're different or not whereas diff() returns a collection of the differences. diff() is like doing a 'git diff' where it tells you everything that is different between the two objects.